voting/src/state.rs

```diff
@@ -27,10 +27,20 @@ pub struct State {
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct TokenManager {
-    pub token_balance: Uint128,
-    // pub locked_tokens: HashMap<u64, Uint128>,
-    //todo map poll_id to weight voted, mainly to find the largest weight staked at withdrawal time
-    pub participated_polls: Vec<u64>, // todo set of polls for the voter
+    pub token_balance: Uint128, // total staked balance
+    pub locked_tokens: HashMap<u64, Uint128>, //maps poll_id to weight voted
```

Comment:
As mentioned above, if you turn this into:

`pub locked_tokens: Vec<(u64, Uint128)>,`

and make use of the [many Vec helpers](https://doc.rust-lang.org/std/vec/struct.Vec.html) this should work as well for you to code and be nicely serializable. It might be a nice addition to `cosmwasm-storage` to make such a type that looks like a `HashMap` but is really just a `Vec` internally.

Reply:
I just found this crate which could drop in if you want to try it. https://contain-rs.github.io/vec-map/vec_map/struct.VecMap.html

(No guarantees, I have never used it, just saw rustdoc and it is similar to what I was saying)